social_core/backends/okta.py

```diff
@@ -0,0 +1,59 @@
+"""
+Okta OAuth2 and OpenIdConnect:
+    https://python-social-auth.readthedocs.io/en/latest/backends/okta.html
+"""
+from six.moves.urllib.parse import urljoin
+
+from ..utils import append_slash
+from .oauth import BaseOAuth2
+
+
+class OktaMixin(object):
+    def api_url(self):
+        return append_slash(self.setting('API_URL'))
+
+    def authorization_url(self):
+        return self._url('v1/authorize')
+
+    def access_token_url(self):
+        return self._url('v1/token')
+
+    def _url(self, path):
+        return urljoin(append_slash(self.setting('API_URL')), path)
+
+    def oidc_config(self):
+        return self.get_json(self._url('/.well-known/openid-configuration?client_id='+self.setting('KEY')))
+
+
+class OktaOAuth2(OktaMixin, BaseOAuth2):
+    """Okta OAuth authentication backend"""
+    name = 'okta-oauth2'
+    REDIRECT_STATE = False
+    ACCESS_TOKEN_METHOD = 'POST'
+    SCOPE_SEPARATOR = ' '
+
```

Comment:
I also had to add the 
`    ID_KEY = "preferred_username"`
to get the UID (via the get_user_id() method) properly set in django social.

Reply:
not user if generic or just for my use case...
and maybe openidconnect need the same ==> move it to the OktaMixin

Reply:
I just checked this using the flask example from: https://github.com/python-social-auth/social-examples. I don't get this same issue.

Reply:
> not user if generic or just for my use case...
> and maybe openidconnect need the same ==> move it to the OktaMixin

In any case this would be fine under OktaOAuth2 as the OpenIdConnect class inherits from the OAuth2 class.

Reply:
I've implemented it anyway.

Reply:
tx! 
to understand the thing, what do you get as get_user_id() when not having ID_KEY defined?

Reply:
I didn't use this personally. I tested it on the example Flask application.